### PR TITLE
extract Update Workflow validation

### PR DESCRIPTION
**Pull request by stephanos**
## What changed?

Extracted the validation of the Start Workflow operation into its own function.

## Why?

To re-use it for the MultiOperation in an upcoming PR.

## How did you test it?

Existing tests.

### CODE REVIEW

File: service/frontend/workflow_handler.go [hunk 1]
```diff
@@ -3403,64 +3403,74 @@ func (wh *WorkflowHandler) UpdateWorkflowExecution(
...
 
 	if request.GetWaitPolicy() == nil {
 		request.WaitPolicy = &updatepb.WaitPolicy{}
 	}
 
-	nsID, err := wh.namespaceRegistry.GetNamespaceID(namespace.Name(request.GetNamespace()))
-	if err != nil {
-		return nil, err
-	}
-
```
Comment (stephanos) [inline service/frontend/workflow_handler.go:3442]: This is the only behavior change. I believe it's safe to remove this from here as I've verified that `EnableUpdateWorkflowExecution` can handle a non-existing namespace.

Comment (tdeebswihart) [reply service/frontend/workflow_handler.go:3442]: Is there a reason we don't want this check in other users of `prepareUpdateWorkflowRequest`?

**Review by tdeebswihart — APPROVED** [review]

File: service/frontend/workflow_handler.go (hunk 1 above)
Comment (tdeebswihart) [reply service/frontend/workflow_handler.go:3442]: I suppose we'll need to grab the nsid in any other consumer regardless

Comment (stephanos) [reply service/frontend/workflow_handler.go:3442]: It's my believe that this should be down higher up in the stack (as it happens here and will for the next use case which is MultiOperation API), ie by the caller. The check is not particular to the Update request, it needs to happen on any kind of request.

Comment (stephanos) [reply service/frontend/workflow_handler.go:3442]: ie it's more for our internal plumbing, not request-specific.

**Review by rodrigozhou — APPROVED** [review]

**Pull request merged**